stdlib/contrib/anaisdg/statsmodels/linearreg.flux

```diff
@@ -43,11 +43,17 @@ import "generate"
 // >     |> statsmodels.linearRegression()
 // ```
 linearRegression = (tables=<-) => {
+    // extract first record of table for later use
+    firstRecord = tables 
+        |> findRecord(
+            fn: (key) => true,
+            idx: 0
+        )
     renameAndSum =
         tables
             |> rename(columns: {_value: "y"})
-            |> map(fn: (r) => ({r with x: 1.0}))
-            |> cumulativeSum(columns: ["x"])
+            // calculate x as difference from first value (in seconds)
+            |> map(fn: (r) => ({ r with x : float(v: uint(v: r._time)  - uint(v: firstRecord._time)) }))
```

Comment:
This method assumes that the first timestamp in the first table in the stream of table should be used as the "origin" time for all other tables. I don't think that's a safe assumption. I think a safer assumption would be to use the `_start` value of the row as the origin timestamp. This time represents the query boundary, or if the data is windowed, the window boundary. This would also negate the need to use `findRecord` at the beginning of the function.

```suggestion
            |> map(fn: (r) => ({ r with x : float(v: uint(v: r._time)  - uint(v: r._start)) }))
```

Reply:
I suppose I'm fine with either, but if you use `r._start`, then your regression will be relative to an arbitrary view point (i.e. the beginning of your window/query), rather than the actual data. Playing around with this a bit, if I use a relative query window (i.e. last thirty days), then the slope of my regression changes (very slightly, but still) every time I run the query, since its calculating relative to the query rather than the actual first data point, even though the data contained within my query window has not changed.

> ...first table in the stream of table should be used as the "origin" time for all other tables. I don't think that's a safe assumption.

I use Influx for very simple data, so I don't think I appreciate the consequences of this. I'm willing to defer to your suggestion, but my gut feeling is that it's not "technically" correct, since the regression should be performed on just the data itself, rather than as a function of the viewport.